js/reference.js

```diff
@@ -1473,6 +1478,49 @@ var ERMrest = (function(module) {
             return newReference;
         },
 
+        /**
+         *
+         * @param {ERMrest.ColumnAggregateFn[]} aggregateList - list of aggregate functions to apply to GET uri
+         * @return {Promise} TODO: not sure of the structure of this returned promise yet
+         */
+        getAggregates: function(aggregateList) {
+            var defer = module._q.defer();
+
+            // use this index to get the next ascii value in the alphabet starting at `a`
+            var aliasCounter = 0;
+            function generateAlias() {
+                var alias = String.fromCharCode(97 + aliasCounter);
+                aliasCounter++;
+                return alias;
+            }
+
+            var url = this.location.service + "/catalog/" + this.location.catalog + "/aggregate/" + this.location.compactPath + "/";
```

Comment:
I think you're still working on this issue. But this is something that feel like I should have emphasised on. `Location.compactPath`, `Location.path`, `Location.uri`, and `Location.compactUri` return uris that chaise understand. To make a request to ermrest use the "ermrset version" of these apis (`Location.ermrestCompactPath`, `Location.ermrstPath`, `Location.ermrestUri`, and `Location.ermrestCompactUri`).

The only difference right now is that ermrest version of these uris will change the `search` pseudo-operator to appropriate operator that ermrest understands.